File: src/TodoListApp.Application/Todo/TodoService.cs
using AutoMapper.Internal.Mappers;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;
using Volo.Abp.Application.Services;
using Volo.Abp.Guids;

namespace TodoListApp.Todo
{
    public class TodoService : ApplicationService
    {
        private readonly TodoManager _todoManager;
        private readonly IGuidGenerator _guidGenerator;
        public TodoService(TodoManager todoManager,IGuidGenerator guidGenerator)
        {
            _todoManager = todoManager;
            _guidGenerator = guidGenerator;
        }

        public async Task<ResponseAPI> GetById(Guid id)
        {
            var result = await _todoManager.GetById(id);
            return new ResponseAPI(true,"",result,"");
        }

        public async Task<ResponseAPI> AddTodo(CreateTodoDto input)
        {
            input.Id = _guidGenerator.Create();
            var todo = ObjectMapper.Map<CreateTodoDto, Todo>(input);
            var result = await _todoManager.AddTodo(todo);
            return new ResponseAPI(true, "", result, "");
        }

        public async Task<ResponseAPI> UpdateTodo(UpdateTodoDto input,Guid id)
        {
            var todo = ObjectMapper.Map<UpdateTodoDto, Todo>(input);
            var result = await _todoManager.UpdateTodo(todo,id);
            return new ResponseAPI(true, "", result, "");
        }

        public async Task<ResponseAPI> DeleteTodo(Guid id)
        {
            var result = _todoManager.DeleteTodo(id);
            return new ResponseAPI(true, "", true, "");
        }

        public async Task<ResponseAPI> GetTodoFilteredListAsync(TodoPagedAndSortedListDto filters)
        {
            var result = await _todoManager.GetTodoList(filters.SearchText, filters.From, filters.To,
                filters.Sorting, filters.Status, filters.Priority);

            var totalCount = await AsyncExecuter.CountAsync(result);
            var sortedList = await AsyncExecuter.ToListAsync(result);
            var pagedList = sortedList.Skip(filters.SkipCount).Take(filters.MaxResultCount).ToList();

            var finalresult = ObjectMapper.Map<List<Todo>, List<TodoDto>>(pagedList);
            return new ResponseAPI(true, "", new {TotalCount = totalCount,Items=finalresult}, "");


        }

    }
}

File: angular/src/app/todo/TodoService.ts
import { RestService } from "@abp/ng.core";
import { Injectable } from '@angular/core';
import { CreateTodoDto, TodoDto, TodoPagedAndSortedListDto } from "./Todo-Models";
import { ResponseAPI } from "../shared/shared.models";

@Injectable({
    providedIn: 'root'
})
export class TodoService {

   apiName = 'Default';
   constructor(private restService: RestService) {}

   GetListFiltered = (filters: TodoPagedAndSortedListDto) => 
   this.restService.request<any, ResponseAPI>({
    method: 'GET',
    url: `/api/app/todo/todo-filtered-list`,
    params: {
        searchText: filters.searchText,
        from: filters.from,
        to: filters.to,
        status: filters.status,
        priority: filters.priority,
        sorting: filters.sorting,
        skipCount: filters.skipCount,
        maxResultCount: filters.maxResultCount,
    },
   },
   { apiName: this.apiName });

   Delete = (id: string) =>
   this.restService.request<any, ResponseAPI>({
    method: 'DELETE',
    url: `/api/app/todo/${id}/todo`,
   },
   { apiName: this.apiName });

   GetById = (id: string) =>
   this.restService.request<any, ResponseAPI>({
    method: 'GET',
    url: `/api/app/todo/${id}/by-id`,
   },
   { apiName: this.apiName });

    Create = (input: CreateTodoDto) =>
    this.restService.request<any, ResponseAPI>({
        method: 'POST',
        url: `/api/app/todo/todo`,
        body: input,
    },
    { apiName: this.apiName });

    Update = (input: CreateTodoDto, id: string) =>
    this.restService.request<any, ResponseAPI>({
        method: 'PUT',
        url: `/api/app/todo/${id}/todo`,
        body: input,
    },
    { apiName: this.apiName });
}